gcp/apis/compute/v1alpha1/network_types.go

```diff
@@ -0,0 +1,331 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+	"net/http"
+
+	googlecompute "google.golang.org/api/compute/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/crossplane/apis/core/v1alpha1"
+)
+
+// NetworkSpec defines the desired state of Network
+type NetworkSpec struct {
+	v1alpha1.ResourceSpec `json:",inline"`
+	SpecForProvider       *GCPNetwork `json:"specForProvider,omitempty"`
+}
+
+// NetworkStatus defines the observed state of Network
+type NetworkStatus struct {
+	v1alpha1.ResourceStatus `json:",inline"`
+	StatusAtProvider        *GCPNetwork `json:"statusAtProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// Network is the Schema for the examplekinds API
+type Network struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkSpec   `json:"spec,omitempty"`
+	Status NetworkStatus `json:"status,omitempty"`
+}
+
+// SetBindingPhase of this ReplicationGroup.
+func (rg *Network) SetBindingPhase(p v1alpha1.BindingPhase) {
+	rg.Status.SetBindingPhase(p)
+}
+
+// SetConditions of this ReplicationGroup.
+func (rg *Network) SetConditions(c ...v1alpha1.Condition) {
+	rg.Status.SetConditions(c...)
+}
+
+// GetBindingPhase of this ReplicationGroup.
+func (rg *Network) GetBindingPhase() v1alpha1.BindingPhase {
+	return rg.Status.GetBindingPhase()
+}
+
+// SetClaimReference of this ReplicationGroup.
+func (rg *Network) SetClaimReference(r *corev1.ObjectReference) {
+	rg.Spec.ClaimReference = r
+}
+
+// GetClaimReference of this ReplicationGroup.
+func (rg *Network) GetClaimReference() *corev1.ObjectReference {
+	return rg.Spec.ClaimReference
+}
+
+// SetClassReference of this ReplicationGroup.
+func (rg *Network) SetClassReference(r *corev1.ObjectReference) {
+	rg.Spec.ClassReference = r
+}
+
+// GetClassReference of this ReplicationGroup.
+func (rg *Network) GetClassReference() *corev1.ObjectReference {
+	return rg.Spec.ClassReference
+}
+
+// SetWriteConnectionSecretToReference of this ReplicationGroup.
+func (rg *Network) SetWriteConnectionSecretToReference(r corev1.LocalObjectReference) {
+	rg.Spec.WriteConnectionSecretToReference = r
+}
+
+// GetWriteConnectionSecretToReference of this ReplicationGroup.
+func (rg *Network) GetWriteConnectionSecretToReference() corev1.LocalObjectReference {
+	return rg.Spec.WriteConnectionSecretToReference
+}
+
+// GetReclaimPolicy of this ReplicationGroup.
+func (rg *Network) GetReclaimPolicy() v1alpha1.ReclaimPolicy {
+	return rg.Spec.ReclaimPolicy
+}
+
+// SetReclaimPolicy of this ReplicationGroup.
+func (rg *Network) SetReclaimPolicy(p v1alpha1.ReclaimPolicy) {
+	rg.Spec.ReclaimPolicy = p
+}
+
+// +kubebuilder:object:root=true
+
+// NetworkList contains a list of Network
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Network `json:"items"`
+}
+
+func init() {
```

Comment:
I don't believe this is needed as it is included in `register.go`.

Reply:
Fixed in the next commit.